### PR TITLE
Add `distTag` option

### LINKED ISSUE #150: Add support for dist-tag versions
Currently, `update-notifier` only support detect the latest version of the package.

The related code is here: https://github.com/yeoman/update-notifier/blob/master/index.js#L102

For [latest-version@4.0.0](https://github.com/sindresorhus/latest-version), it already supports dist-tag versions. But it will be a breaking change because [latest-version@4.0.0](https://github.com/sindresorhus/latest-version) requires Node.js v6 or later.

I think the API can be designed like this:

Add a `distTag` option and default to `latest`.

```javascript
const updateNotifier = require('update-notifier');
const pkg = require('./package.json');

const notifier = updateNotifier({pkg, distTag: 'beta'});

notifier.notify();
```

### PR DESCRIPTION
**Pull request by LitoMore**
Resolve #150

### CODE REVIEW

**Comment by LitoMore** [comment]
@sindresorhus Could you help me review why this CI failed?

Should I update the `.travis.yml` for higher Node.js version in this pull request?

**Comment by LitoMore** [comment]
@SBoudrias ping

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 language: node_js
 node_js:
   - '6'
-  - '4'
+  - '8'
+  - '10'
+  - 'node'
```
Comment (sindresorhus) [inline .travis.yml]: Don't do unrelated changes.

File: index.js [hunk 2]
```diff
@@ -36,7 +37,7 @@ class UpdateNotifier {
 		this.packageVersion = options.pkg.version;
 		this.updateCheckInterval = typeof options.updateCheckInterval === 'number' ? options.updateCheckInterval : ONE_DAY;
 		this.hasCallback = typeof options.callback === 'function';
-		this.callback = options.callback || (() => {});
+		this.callback = options.callback || (() => { });
```
Comment (sindresorhus) [inline index.js]: Don't do unrelated changes.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by LitoMore** [comment]
@sindresorhus Code updated.
